src/app/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,72 @@
+import { useNewsContext } from '@/context/news.context';
+import React from 'react';
+import classNames from 'classnames/bind';
+import getPaginationRange from '@/utils/getPaginationRange';
+import { useSearchParams, useRouter } from 'next/navigation';
+import styles from './Pagination.module.scss';
+
+type Props = {
+  currentPage: number;
+  setCurrentPage(value: number): void;
+};
+
+const cx = classNames.bind(styles);
+
+function Pagination({ currentPage, setCurrentPage }: Props) {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const { news } = useNewsContext();
+  const totalPageNumber = Math.floor(100 / news.length); // 100 articles - API limit
+  const isPreviousBtnDisabled = currentPage === 1;
+  const isNextBtnDisabled = currentPage === totalPageNumber;
+
+  const paginationElements = getPaginationRange(totalPageNumber, currentPage);
+
+  const handlePageChange = (value: number) => {
+    setCurrentPage(Math.max(1, Math.min(totalPageNumber, value)));
+    const query = `?q=${searchParams.get('q')}&page=${value}`;
```

Comment:
Please move it outside of this component. If you put it here it will be impossible to reuse this component somewhere else. The main idea behind component is that they should be really stupid and only do one thing without knowing anything where they will be used